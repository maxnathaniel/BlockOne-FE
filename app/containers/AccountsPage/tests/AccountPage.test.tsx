import React from 'react';
import { queryByText, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import AccountPage from '../index';

const accountId = '13535hr-35afa2-25252fa-26396ahb';
const defaultState = {
  accounts: {
    'Block One Savings': { amount: 100000, id: accountId, type: 'Savings', isSelected: true, transactions: [] },
  },
  userConfig: { selectedAccount: 'Block One Savings' }
};

jest.mock('react-redux', () => ({
  useSelector: () => defaultState,
  useDispatch: () => jest.fn(),
}));

describe('<AccountPage />', () => {
  it('should render the various inner Elements', () => {
    const { container, getByText } = render(<AccountPage />);
    expect(getByText('Welcome back!')).toBeInTheDocument();
    expect(getByText('Create a new account by clicking on the button below')).toBeInTheDocument();

    const modalHeader = queryByText(container, 'Add Account');
    const accountNameLabel = queryByText(container, 'Account Name');
    const accountTypeLabel = queryByText(container, 'Account Type');
    const accountCurrency = queryByText(container, 'Account Currency');
    expect(modalHeader).not.toBeInTheDocument();
    expect(accountNameLabel).not.toBeInTheDocument();
    expect(accountTypeLabel).not.toBeInTheDocument();
    expect(accountCurrency).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should open Add Account Modal on click of Add Btn', () => {
    const { container, getByText, getAllByText, getByTestId } = render(<AccountPage />);

    const btn = getByText('Create New Account');
    btn.click();

    expect(getAllByText('Add Account')).toHaveLength(2);
    expect(getByText('Account Name')).toBeInTheDocument();
    expect(getByText('Account Type')).toBeInTheDocument();
    expect(getByText('Account Currency')).toBeInTheDocument();
    expect(getByTestId('account-name')).toBeInTheDocument();
    expect(getByTestId('account-dropdown')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});