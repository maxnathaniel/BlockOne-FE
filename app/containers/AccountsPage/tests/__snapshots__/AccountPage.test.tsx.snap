// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountPage /> should open Add Account Modal on click of Add Btn 1`] = `
<div>
  <div
    class="sc-AykKG OdUwQ"
  >
    <div
      class="sc-AykKC ePKNZv"
    >
      <h3
        class="sc-AykKD cYEpnF"
      >
        Welcome back!
      </h3>
      <p
        class="sc-AykKE bkDhgi"
      >
        Create a new account by clicking on the button below
      </p>
      <div
        class="sc-AykKF jvvefI"
      >
        <button
          class="ui teal medium button"
          name="Create New Account"
        >
          Create New Account
        </button>
      </div>
    </div>
    <div
      class="sc-AykKH hLQGxy"
    >
      <div
        class="sc-fzXfLO dZfGKo"
      >
        <div
          class="sc-fzXfLQ iprOSq"
        >
          <p
            class="sc-fzXfLW iASfNc"
          >
            13535hr-35afa2-25252fa-26396ahb
          </p>
        </div>
        <div
          class="sc-fzXfLR jCJQRO"
        >
          <h4
            class="sc-fzXfLT ipalgS"
          >
            Block One Savings
          </h4>
          <p
            class="sc-fzXfLU jtdEZA"
          >
            Savings
             account
          </p>
        </div>
        <div
          class="sc-fzXfLS lflAaF"
        >
          <p
            class="sc-fzXfLV iKPxLq"
          >
            $100,000.00
          </p>
          <i
            aria-hidden="true"
            class="send icon"
            data-testid="send-icon"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AccountPage /> should render the various inner Elements 1`] = `
<div>
  <div
    class="sc-AykKG OdUwQ"
  >
    <div
      class="sc-AykKC ePKNZv"
    >
      <h3
        class="sc-AykKD cYEpnF"
      >
        Welcome back!
      </h3>
      <p
        class="sc-AykKE bkDhgi"
      >
        Create a new account by clicking on the button below
      </p>
      <div
        class="sc-AykKF jvvefI"
      >
        <button
          class="ui teal medium button"
          name="Create New Account"
        >
          Create New Account
        </button>
      </div>
    </div>
    <div
      class="sc-AykKH hLQGxy"
    >
      <div
        class="sc-fzXfLO dZfGKo"
      >
        <div
          class="sc-fzXfLQ iprOSq"
        >
          <p
            class="sc-fzXfLW iASfNc"
          >
            13535hr-35afa2-25252fa-26396ahb
          </p>
        </div>
        <div
          class="sc-fzXfLR jCJQRO"
        >
          <h4
            class="sc-fzXfLT ipalgS"
          >
            Block One Savings
          </h4>
          <p
            class="sc-fzXfLU jtdEZA"
          >
            Savings
             account
          </p>
        </div>
        <div
          class="sc-fzXfLS lflAaF"
        >
          <p
            class="sc-fzXfLV iKPxLq"
          >
            $100,000.00
          </p>
          <i
            aria-hidden="true"
            class="send icon"
            data-testid="send-icon"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
